File: examen_java/src/main/java/com/groupeisi/examen_java/students/services/StudentService.java
package com.groupeisi.examen_java.students.services;

import com.groupeisi.examen_java.students.dto.requests.StudentDtoRequest;
import com.groupeisi.examen_java.students.dto.responses.StudentDtoResponse;

import java.util.List;
import java.util.Optional;

public interface StudentService {
    Optional<StudentDtoResponse>saveStudent(StudentDtoRequest studentDto);
    Optional<List<StudentDtoResponse>> getAllStudents();
    Optional<StudentDtoResponse>getStudentById(Long id);
    boolean deleteStudent(Long id);
    Optional<StudentDtoResponse>updateStudent(Long id, StudentDtoRequest studentDtoRequest);

}

File: front-exam/src/app/services/student/student.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { StudentDtoRequest, StudentDtoResponse } from '../../models/student.dto';


@Injectable({
  providedIn: 'root',
})
export class StudentService {
 
  private apiUrl = 'http://localhost:8083/api/students'; 

  constructor(private http: HttpClient) {}


  registerStudent(studentData: StudentDtoRequest): Observable<StudentDtoResponse> {
    return this.http.post<StudentDtoResponse>(this.apiUrl, studentData);
  }

  getAllStudents(): Observable<StudentDtoResponse[]> {
    return this.http.get<StudentDtoResponse[]>(this.apiUrl);
  }


  deleteStudent(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  updateStudent(id: number, student: StudentDtoRequest): Observable<StudentDtoResponse> {
    return this.http.put<StudentDtoResponse>(`${this.apiUrl}/${id}`, student);
  }
}